src/test/ui/asm/naked-functions.rs

```diff
@@ -132,38 +143,89 @@ unsafe extern "C" fn invalid_options_continued() {
 }
 
 #[naked]
+#[inline(never)]
 pub unsafe fn default_abi() {
     //~^ WARN Rust ABI is unsupported in naked functions
     //~| WARN this was previously accepted
     asm!("", options(noreturn));
 }
 
 #[naked]
+#[inline(never)]
 pub unsafe extern "Rust" fn rust_abi() {
     //~^ WARN Rust ABI is unsupported in naked functions
     //~| WARN this was previously accepted
     asm!("", options(noreturn));
 }
 
 #[naked]
+#[inline(never)]
 pub extern "C" fn valid_a<T>() -> T {
     unsafe { asm!("", options(noreturn)); }
 }
 
 #[naked]
+#[inline(never)]
 pub extern "C" fn valid_b() {
     unsafe { { {
         asm!("", options(noreturn)); ; ; ;
     } ; }  ; }
 }
 
 #[naked]
+#[inline(never)]
 pub unsafe extern "C" fn valid_c() {
     asm!("", options(noreturn));
 }
 
 #[cfg(target_arch = "x86_64")]
 #[naked]
+#[inline(never)]
 pub unsafe extern "C" fn valid_att_syntax() {
     asm!("", options(noreturn, att_syntax));
 }
+
+#[naked]
+//~^ WARN naked functions require #[inline(never)]
```

Comment:
The tracking issue says that this should be an error, but it's a warning in this PR. Is this just a transitional lint to avoid breaking nightly code? If so, seems reasonable to me.

Reply:
@josephlr There's a bunch of these lints that should be errors according to the RFC. I wanted to get them all in as warnings first and then file a follow-up which switches the relevant ones to errors. This allows us to reason about them as an entire set.